org.eclipse.jdt.core.compiler.batch/src/org/eclipse/jdt/internal/compiler/lookup/TypeBinding.java

```diff
@@ -1765,6 +1765,14 @@ public void exitRecursiveFunction() {
 public boolean isFunctionalType() {
 	return false;
 }
+
+/**
+ * Returns true if the type or one of its constituents is a SourceTypeBinding that has not yet connected its hierarchy
+ */
+public boolean hasUnconnectedSourceHierarchy() {
```

Comment:
I was tempted to use (and extend) the existing `isHierarchyConnected()`, but specifically regarding `BinaryTypeBinding` the requirements seem to be incompatible.

Reply:
this method no longer exists